app/src/main/java/org/thoughtcrime/securesms/conversation/v2/dialogs/DownloadDialog.kt

```diff
@@ -1,54 +1,60 @@
 package org.thoughtcrime.securesms.conversation.v2.dialogs
 
 import android.app.Dialog
-import android.graphics.Typeface
 import android.os.Bundle
-import android.text.Spannable
-import android.text.SpannableStringBuilder
-import android.text.style.StyleSpan
 import androidx.fragment.app.DialogFragment
 import com.squareup.phrase.Phrase
 import dagger.hilt.android.AndroidEntryPoint
 import network.loki.messenger.R
+import org.session.libsession.database.StorageProtocol
 import org.session.libsession.messaging.contacts.Contact
-import org.session.libsession.messaging.jobs.AttachmentDownloadJob
 import org.session.libsession.messaging.jobs.JobQueue
+import org.session.libsession.messaging.sending_receiving.attachments.DatabaseAttachment
 import org.session.libsession.utilities.recipients.Recipient
 import org.session.libsession.utilities.StringSubstitutionConstants.CONVERSATION_NAME_KEY
 import org.thoughtcrime.securesms.createSessionDialog
 import org.thoughtcrime.securesms.database.SessionContactDatabase
-import org.thoughtcrime.securesms.dependencies.DatabaseComponent
+import org.thoughtcrime.securesms.util.createAndStartAttachmentDownload
 import javax.inject.Inject
 
 /** Shown when receiving media from a contact for the first time, to confirm that
  * they are to be trusted and files sent by them are to be downloaded. */
 @AndroidEntryPoint
-class DownloadDialog(private val recipient: Recipient) : DialogFragment() {
+class AutoDownloadDialog(private val threadRecipient: Recipient,
+                     private val databaseAttachment: DatabaseAttachment
+) : DialogFragment() {
 
+    @Inject lateinit var storage: StorageProtocol
     @Inject lateinit var contactDB: SessionContactDatabase
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog = createSessionDialog {
-        val accountID = recipient.address.toString()
-        val contact = contactDB.getContactWithAccountID(accountID)
-        val name = contact?.displayName(Contact.ContactContext.REGULAR) ?: accountID
-
+        val threadId = storage.getThreadId(threadRecipient) ?: run {
+            dismiss()
+            return@createSessionDialog
+        }
+
+        val displayName = when {
+            threadRecipient.isCommunityRecipient -> storage.getOpenGroup(threadId)?.name ?: "UNKNOWN"
+            threadRecipient.isLegacyGroupRecipient -> storage.getGroup(threadRecipient.address.toGroupString())?.title ?: "UNKNOWN"
+            threadRecipient.isGroupV2Recipient -> threadRecipient.name ?: "UNKNOWN"
+            else -> storage.getContactWithAccountID(threadRecipient.address.serialize())?.displayName(Contact.ContactContext.REGULAR) ?: "UNKNOWN"
+        }
```

Comment:
What is this hardcoded "UNKNOWN"? I don't think we should allow to ever have a state where the display name is a hardcoded english string.

Reply:
I have changed to use "Unknown sender" from our string resource let me know if you have a better alternative

Reply:
I've updated to use the short string instead